server/helpers/ffmpeg-utils.ts

```diff
@@ -1,10 +1,47 @@
+import * as fs from 'fs'
 import * as ffmpeg from 'fluent-ffmpeg'
 import { join } from 'path'
 import { VideoResolution } from '../../shared/models/videos'
-import { CONFIG, VIDEO_TRANSCODING_FPS } from '../initializers'
+import { CONFIG, VIDEO_TRANSCODING_FPS, FFMPEG_NICE } from '../initializers'
 import { unlinkPromise } from './core-utils'
 import { processImage } from './image-utils'
 import { logger } from './logger'
+import { checkFFmpegEncoders } from '../initializers/checker'
+
+async function isVideoFileStreamable (path: string): Promise<{ [key: string]: boolean }> {
+  const fileSizeInBytes = fs.statSync(path).size
+  const fileSizeToRead = Math.max(10000, fileSizeInBytes * 0.5)
+  const offset = Math.min(fileSizeToRead, fileSizeInBytes)
+  // basically we want to look at the first 5% of the file, with a minimum of 10000 bytes (if the file is big enough)
+  let stream = fs.createReadStream(path, {
+    encoding: null,
+    start: 0,
+    end: offset
+  })
+
+  const isAVC = await getVideoFileStream(path).then(s => {
+    return s['codec_name'] === 'h264' &&
+           s['bit_rate'] < 3000000
+  })
+  const isAAC = await audio.get(ffmpeg, path).then(s => {
+    return s['codec_name'] === 'aac'
+  })
+  console.log('got it until aac')
+  const hasFastStart = await new Promise<boolean>((res, rej) => {
+    ffmpeg(path, { stdoutLines: 0 })
```

Comment:
You should change option to :
You forgot ' for stdoutLines (I'm not totally sure that is a must but it should be here) and niceness is better except if that only for check ffmpeg instalation
```
{ 'niceness': FFMPEG_NICE.THUMBNAIL, 'stdoutLines': 0 }
```

Reply:
@Jorropo unless an attribute has spaces or special characters, there shouldn't be a need to quote it.

As for the niceness, it's not helping the tests, nor is actually interesting in production since this is a short-lived command.